### PR TITLE
Feature GitHub actions ci

### LINKED ISSUE #572: Migrate CI from travis to github actions?
Would it be a good idea to move away from travis CI, and use GHA instead? It doesn't look like free open source support can be guaranteed at travis for the foreseeable future.

I've noted a couple of lintr users have raised issues re using lintr on GHA, but i haven't any experience using GHA yet.

### PR DESCRIPTION
**Pull request by dragosmg**
I believe this can be merged now. 
~Still a WIP, do not merge yet.~
Added CI with GitHub Actions. Travis CI ~not~ removed ~yet~. closes #572

### CODE REVIEW

**Comment by dragosmg** [comment]
@russHyde would you be able to convert this PR to draft PR? thanks

**Comment by russHyde** [comment]
Don't worry, if it has WIP in the title it won't be merged. Thanks for your work in this.

**Comment by russHyde** [comment]
Once converted from Travis-based to GHA-based CI, will the code-coverage reports will be appended to the existing codecov webpage?

**Comment by dragosmg** [comment]
@russHyde that's a very good question. I think so. I don't have experience with this specific situation (converting from Travis + Codecov to GitHub Actions + Codecov), but if I read the `yml` files correctly, they're doing the same thing, namely running `covr::codecov()`, which should upload the result to the same codecov.io page. Given the structure of the URL (`https://codecov.io/github/jimhester/lintr?branch=master`) I don't foresee any changes. 
Not sure how to best test this though. Maybe @jimhester has some ideas.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # lintr
+[![R build status](https://github.com/jimhester/lintr/workflows/R-CMD-check/badge.svg)](https://github.com/jimhester/lintr/actions)
 [![Travis-CI Build Status](https://travis-ci.org/jimhester/lintr.svg?branch=master)](https://travis-ci.org/jimhester/lintr)
```
Comment (russHyde) [inline README.md:3]: Remember to remove the Travis badge when you take away `.travis.yml`

**Review by russHyde — CHANGES_REQUESTED** [review]
Thanks for this. Could you 1) remove .travis.yml; 2) add a note to NEWS.md (`<a quick description of the change> (#issue/PR-numbers, <at>your-github-ID)`); and we'll try to get this into master this week

File: README.md (hunk 1 above)
Comment (dragosmg) [reply README.md]: done

**Comment by dragosmg** [comment]
Once the checks are completed, this is good to go. Checks on Windows will still fail if this is merged before #576. I did a test on an old package of mine which had Travis, Appveyor and Codecov and migrated to GH Actions + Codecov. Things seem to be working fine. One thing you might need to do is go to Settings > Webhooks and remove any stale webhooks once the switch is done.

**Comment by AshesITR** [comment]
I'll merge #576 today

**Comment by AshesITR** [comment]
@russHyde I think we should merge this asap, the travis situation is becoming a major hindrance in getting PRs merged.
Unfortunately travis seems to not even report build status for this PR currently.

At least it seems that two of the six required builds have already passed:
https://travis-ci.org/github/jimhester/lintr/builds/746060434

**Review by russHyde — DISMISSED** [review]
LGTM

**Comment by russHyde** [comment]
I'm not sure how to override the legacy checks. The Travis checks haven't completed because checking on Travis was removed in this PR, but checking on Travis is a "Required status". Is there a setting that we can get @jimhester to modify to permit the migration?

**Comment by AshesITR** [comment]
screenshot

That looks like the appropriate setting.

**Comment by dragosmg** [comment]
> @russHyde I think we should merge this asap, the travis situation is becoming a major hindrance in getting PRs merged.
> Unfortunately travis seems to not even report build status for this PR currently.

I've realised these days the Travis situation is really bad. Let's hope his solves it and adds checks on macOS and Windows too. Checks should now run in minutes rather than days.

**Comment by dragosmg** [comment]
@russHyde & @AshesITR any news on this PR? Are you still aiming to get this merged this weekend?

**Comment by AshesITR** [comment]
We need @jimhester because the CI settings need to be tweaked before we can merge.

**Comment by dragosmg** [comment]
Oh, I see. I was under the impression the was at least one other person with more privileges.

**Comment by AshesITR** [comment]
In the mean time we've kicked off a delinting project for lintr (See #584).
If you want to, you can include `lintr::lint_package()` in the GHA pipeline as well.

**Comment by dragosmg** [comment]
@AshesITR I missed the delinting conversation for some reason. I can add another workflow (similar to the test-coverage one) that runs `lintr::lint_package()`. Do you want me to do that?

**Comment by AshesITR** [comment]
That would be very cool.

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        with:
+          path: ${{ env.R_LIBS_USER }}
+          key: ${{ runner.os }}-${{ hashFiles('.github/R-version') }}-1-${{ hashFiles('.github/depends.Rds') }}
+          restore-keys: ${{ runner.os }}-${{ hashFiles('.github/R-version') }}-1-
+
+      - name: Install dependencies
+        run: |
+          install.packages(c("remotes"))
+          remotes::install_deps(dependencies = TRUE)
+          remotes::install_cran("lintr")
```
Comment (AshesITR) [inline .github/workflows/lint.yaml:41]: Can we install the currently built version somehow?
CRAN lintr doesn't support e.g. directory exclusions, which we use extensively now.

**Review by AshesITR — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yaml (hunk 2 above)
Comment (dragosmg) [reply .github/workflows/lint.yaml]: Done. I've run a test on a toy package and directory exclusions seem to work properly with the dev version of {lintr}

**Comment by dragosmg** [comment]
See test run (https://github.com/dragosmg/dmgr/actions/runs/390375711). If `tests` directory is excluded then there is no lint. Also the lintr-bot doing its thing (https://github.com/dragosmg/dmgr/commit/82e1670b93ed3dfb4a120d2c5638531e1614157f)

**Review by AshesITR — APPROVED** [review]
LGTM

**Review by AshesITR — APPROVED** [review]

**Pull request merged**